### PR TITLE
Jonmv/async doc v1 implementation

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs please review. Will add tests as they're ready. Note that first 5 commits are already in #14550

### CODE REVIEW

**Comment by jonmv** [comment]
Hmm, last commit wasn't good. Should use `Condition` for this.

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/DocumentOperationExecutor.java [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+                    // Ready for retry — remove from queue and send.
+                    if (operation.readyAt().isBefore(clock.instant())) {
+                        action.accept(operation);
+                        operations.remove();
+                        continue;
+                    }
+                    // Not yet ready for retry — track earliest time to wake up again.
+                    waitUntilMillis = Math.min(waitUntilMillis, operation.readyAt.toEpochMilli());
+                }
+                synchronized (queue) {
```
Comment (bjorncs) [inline vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/DocumentOperationExecutor.java:264]: Do not use a concurrent collection as an external monitor (unless `ConcurrentLinkedQueue` says it's okay).

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+        this.executor = executor;
+        this.parser = parser;
+        this.metric = metric;
+        this.metrics = new DocumentApiMetrics(metricReceiver, "documentV1");
+        this.handlers = defineApi();
+    }
+
+    @Override
+    public ContentChannel handleRequest(Request rawRequest, ResponseHandler rawResponseHandler) {
+        metric.add("handled.requests", 1, HandlerMetricContextUtil.contextFor(rawRequest, metric, getClass()));
```
Comment (bjorncs) [inline vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java:145]: Consider moving the metric aggregation to `HandlerMetricContextUtil` and replace this with a e.g `HandlerMetricContextUtil.onHandledRequest(...)`.

**Review by bjorncs — APPROVED** [review]

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/DocumentOperationExecutor.java (hunk 1 above)
Comment (jonmv) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/DocumentOperationExecutor.java]: Good point. Already changed to allow tests to notify maintainers, though :)

**Comment by jonmv** [comment]
I think we're almost ready to roll. May as well have this merged. Only need some tests for the REST API handler now!

**Pull request merged**